modules/network/service-endpoint-policies/main.bicep

```diff
@@ -42,6 +53,28 @@ resource serviceEndpointPolicy 'Microsoft.network/ServiceEndpointPolicies@2022-0
   }
 }
 
+resource serviceEndpointPolicy_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock)) {
+  name: '${serviceEndpointPolicy.name}-${lock}-lock'
+  properties: {
+    level: any(lock)
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: serviceEndpointPolicy
+}
+
+module serviceEndpointPolicy_roleAssignments '.bicep/nested_roleAssignments.bicep' = [for (roleAssignment, index) in roleAssignments: {
```

Comment:
@tyconsulting thanks a lot for taking the time to go through all requests. You did an incredible job here! 

As a last thing: could you please add locks and rbac to the common test and regenerate readmes? Then the module would be complete. Thanks!

Reply:
Approving for now, missing validation will be integrated in the common test. CC: @ahmadabdalla